README.md

```diff
@@ -1,31 +1,65 @@
-docker base
-===========
+# docker base
 
 This role installs baseline needs for machines running Docker, beyond the Docker Engine itself,
 including:
 
  -  Requirements for running Ansible's `docker` module
- -  Requirements for running Docker containers via `upstart` (optional)
  -  Utilities for garbage collecting unused Docker resources
  -  Customized configurations of the Docker Engine
- -  Installs docker-compose
+ -  Installs docker-compose (optional)
  -  Pulls some common images that we anticipate using in many circumstances
 
-Role Variables
---------------
+## Role Variables
 
 This role expects the following variables:
 
- -  `docker_log_rotate_interval` controls the frequency of Docker log rotation
- -  `docker_log_rotate_count` controls the number of Docker logs to retain
- -  `docker_uses_upstart` defines whether `upstart` support will be installed
- -  `docker_py_version` defines the version of the Docker Python library being used
- -  `docker_compose_version` defines the version of docker-compose to install (defaults to newest)
+ - `docker_version`: the version of Docker that is installed on the target host. Note that docker
```

Comment:
Depends how you define "more straightforward"...

If you're just applying our "docker" and "docker-base" roles in sequence, your deploy project will have set "docker_version" so that the former role would do the right thing, and the precondition for the latter role will always be automatically satisfied.

We could add a check to the docker-base role. It would complicate the code, slow down all deploys, and only be relevant when we do not apply the two roles in sequence.

There's also the weird scenario where docker_version is set, the playbook applies the two roles in sequence, but tags prevent the running of the first role. In that case, relying on "docker_version" might cause docker-base to do the wrong thing.

In practice, I expect most folks to set docker_version, docker_py_version, and docker_compose_version to inter-compatible version numbers. Given the havoc that can occur if those versions don't line up, there's an argument for having docker-base fail if docker_version doesn't match what's installed.